llama.cpp

```diff
@@ -5019,6 +5025,11 @@ static void llm_load_print_meta(llama_model_loader & ml, llama_model & model) {
         LLAMA_LOG_INFO("%s: expert_weights_scale = %.1f\n",   __func__, hparams.expert_weights_scale);
         LLAMA_LOG_INFO("%s: rope_yarn_log_mul    = %.4f\n",   __func__, hparams.rope_yarn_log_mul);
     }
+
+    if (model.arch == LLM_ARCH_QWEN2MOE) {
+        LLAMA_LOG_INFO("%s: n_ff_exp         = %d\n",     __func__, hparams.n_ff_exp);
+        LLAMA_LOG_INFO("%s: n_ff_shared_exp  = %d\n",     __func__, hparams.n_ff_shared_exp);
+    }
```

Comment:
Ooof, this is broken!

`llm_load_print_meta` is called before `llm_load_tensors` where these params are set, so will always print 0.

It is not a problem introduced in this PR though, this affects others too, like `DeepSeekV2`.

Reply:
That's very strange. When I run `./main` to test & debug, i get the following log snippet:
```
llm_load_print_meta: general.name     = Qwen2-57B-A14B-Instruct
llm_load_print_meta: BOS token        = 151643 '<|endoftext|>'
llm_load_print_meta: EOS token        = 151645 '<|im_end|>'
llm_load_print_meta: PAD token        = 151643 '<|endoftext|>'
llm_load_print_meta: LF token         = 148848 'ÄĬ'
llm_load_print_meta: EOT token        = 151645 '<|im_end|>'
llm_load_print_meta: n_ff_exp         = 2560
llm_load_print_meta: n_ff_shared_exp  = 20480
```

suggesting that the data is loaded before printing

Reply:
Never mind me, I'm talking nonsense :)